### PR TITLE
Dataset and repository delete cron fixes

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java [hunk 1]
```diff
@@ -771,7 +629,9 @@ private void deleteRepositories(Session session) {
         } catch (Exception ex) {
           LOGGER.warn("DeleteEntitiesCron : deleteRepositories : Exception: ", ex);
         } finally {
-          transaction.commit();
+          if (transaction != null) {
+            transaction.commit();
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java]: We should not commit something that had an exception during execution.

**Review by conradoverta — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java]: FIxed

**Review by conradoverta — APPROVED** [review]
Approved assuming the build passes.

Could you also double check the other cron jobs to make sure we're not committing something that has failed? I imagine you copied this skeleton from somewhere else.

**Review by coryverta — CHANGES_REQUESTED** [review]
@ad-47 the build is failing due to compilation errors

**Review by coryverta — APPROVED** [review]

**Pull request merged**